testing/mocker/create_common.py

```diff
@@ -0,0 +1,150 @@
+        modality_category: str = "anat",
+    ):
+        raise NotImplementedError
+
+    def create_file_stem(self, session: str or None, bids_details: dict):
+        """
+        Create a file stem for a nifti/JSON image pairs.
+        :param session:
+        :param bids_details:
+        :return:
+        """
+
+        if session:
+            stem = f"sub-{self.index_subject:02d}_ses-{session:02d}"
+        # No session spec
+        else:
+            stem = f"sub-{self.index_subject:02d}"
+
+        # Append all possible bids keywords
+        for keyword in bids_details.keys():
+            if "MODALITY" in keyword:
+                stem += f"_{bids_details.get('MODALITY', '')}"
+            elif "LABELS" in keyword:
+                stem += f"_{bids_details.get('LABELS', '')}"
+            else:
+                stem += f"_{keyword}-{bids_details.get(keyword,'')}"
+        return stem
+
+    def reject_missing_modalities(self, list_subject_specific_bids_dicts: List[dict]):
+        """
+        Validate the list of anatomy modalities.
+        :param list_subject_specific_bids_dicts:
+        :return:
+        """
+        for subject_specific_bids_dict in list_subject_specific_bids_dicts:
+            if "MODALITY" not in subject_specific_bids_dict:
+                logger.critical(f"MODALITY is not present in the subject dictionary provided: {pprint.pformat(subject_specific_bids_dict)}")
+                raise ValueError(
+                    f"Missing MODALITY field in the provided subject_specific_bids_dict."
+                )
+    def reject_missing_labels(self, list_subject_specific_bids_dicts: List[dict]):
+        """
+        Validate the list of anatomy modalities.
+        :param list_subject_specific_bids_dicts:
+        :return:
+        """
+        for subject_specific_bids_dict in list_subject_specific_bids_dicts:
+            if "LABELS" not in subject_specific_bids_dict:
+                logger.critical(f"LABELS is not present in the subject dictionary provided: {pprint.pformat(subject_specific_bids_dict)}")
+                raise ValueError(
+                    f"Missing LABELS field in the provided subject_specific_bids_dict."
+                )
+    def generate_file_description_dict(self, bids_detail: dict) -> dict:
+        """
+        Fill a json dictionary with a description of the nifti image.
+        :param bids_detail:
+        :return:
+        """
+        mock_meta_data = {
+            "Modality": "MR",
+            "MagneticFieldStrength": 3,
+            "Manufacturer": "MOCK",
+            "InstitutionName": "MOCK Research Center",
+            "MRAcquisitionType": "3D",
+            "SeriesDescription": bids_detail.get("MODALITY", ""),
+            "ProtocolName": bids_detail.get("MODALITY", ""),
+            "EchoTime": 1,
+            "RepetitionTime": 1,
+            "InversionTime": 1,
+            "FlipAngle": 1,
+            "SliceThickness": 1,
+            "ConversionSoftware": "MOCK",
```

Comment:
```suggestion
            "MagneticFieldStrength": 3,
            "Manufacturer": "MOCK",
            "InstitutionName": "MOCK Research Center",
            "MRAcquisitionType": "3D",
            "EchoTime": 1,
            "RepetitionTime": 1,
            "InversionTime": 1,
            "FlipAngle": 1,
            "ConversionSoftware": "MOCK",
```

As mentioned earlier, supported BIDS JSON fields depends on the data type.
In the sense that you _can_ have additional field... but you're not _supposed_ to. In the future, it is possible that the BIDS validator will reject those files, so I recommend having compliant JSON.

That means that each data type we use (`anat`, `micr`, etc) would need it's own set on JSON field, and we have to make sure its value is of the right type.
(I don't suggest that the value is realistic, only the right type (`str`, `int`, etc).

Here I removed the fields that are not supported by `anat`.

Reply:
Great point. @konantian has done a great job generating some of the mocking JSON files with place holder mock values for more authentic mocking experience. This is now incorporated as part of the mocerk.